src/app/reporting/Engine.cpp

```diff
@@ -317,6 +355,9 @@ CHIP_ERROR Engine::ScheduleRun()
 
 void Engine::Run()
 {
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+    ScopedLock lock(*this);
```

Comment:
Why do we need lock here since this function should be called in main loop.

Reply:
We may create the seperate PR for lock?

Reply:
We can create a separate PR, however that should be in master or a separate commit on top of this one.